source/slang/slang-options.cpp

```diff
@@ -1655,7 +1655,15 @@ SlangResult OptionsParser::addEmbeddedLibrary(const CodeGenTarget format, Compil
         m_rawTargets.add(rawTarget);
     }
 
-    m_rawTargets.getLast().optionSet.add(option, true);
+    getCurrentTarget()->optionSet.add(option, true);
+    getCurrentTarget()->optionSet.add(CompilerOptionName::GenerateWholeProgram, true);
```

Comment:
Setting the GenerateWholeProgram flag option here negates the need for the "if lib_6_6" hack that we've been talking about.